Drive/rover_drive_system.hpp

```diff
@@ -34,63 +44,130 @@ class RoverDriveSystem
+    // GET /drive JSON value ?
+    if (true)  // change 'true' to something for localhost connection found
+    {
+      char response[] = {};
+      ParseMissionControlData(response);
+      Move(mission_control_data_.drive_mode,
+           mission_control_data_.rotation_angle, mission_control_data_.speed);
+      return true;
+    }
+    sjsu::LogError("Unable to reach mission control server");
+    return false;
+  }
+
+  /// Main function for handling all the rover drive system functionality.
+  /// @param mode Updates drive mode if new mode is different from current.
+  /// @param rotation_angle adjusts wheel position depending on mode.
+  /// @param speed adjusts the movement speed of the rover.
+  void Move(char mode, float rotation_angle, float speed)
+  {
+    if (mission_control_data_.is_operational)
     {
-      // checks if the mode is different than what is currently running
       if (mode != static_cast<char>(current_mode_))
       {
         SetMode(mode);
       }
-      HandleModeMovement(rotation_angle, speed);
+      HandleRoverMovement(rotation_angle, speed);
       UpdateMissionControlData();
     }
   };
 
-  // Homes all the wheels so the motors know their actual position. Returns true
-  // if successful.
-  bool Home()
+  /// Resets all the wheels so the motors know their actual position.
+  /// @return true if successfully resets wheels into start position
+  bool Reset()
   {
+    SetMode();
     return true;
   };
 
-  bool rover_is_operational_ = false;
-
  private:
-  ///  Sets the new mode for the rover. Will reduce rover movement speed to zero
-  ///  before changing wheel mode & tire alignment
-  void SetMode(char mode)
+  /// Sets all wheels to the speed provided. Wheel class handles max/min speeds
+  /// @param speed the new movement speed of the rover
+  void SetWheelSpeed(units::angular_velocity::revolutions_per_minute_t speed)
+  {
+    left_wheel_.SetSpeed(speed);
+    right_wheel_.SetSpeed(speed);
+    back_wheel_.SetSpeed(speed);
+  };
+
+  /// Updates Mission Control /drive/status endpoint with rover's current status
+  void UpdateMissionControlData()
+  {
+    if (GetRoverData())
+    {
+      // POST, http://localhost:3000/drive/status, JSON.stringify(rover_data_)
+    }
+    else
+    {
+      sjsu::LogError("Unable to retrieve wheel data!");
+    }
+  };
+
+  /// Gets the speed and position/angle of each wheel on the rover
+  /// @return true if rover is able to retrieve data from all the wheels
+  bool GetRoverData()
+  {
+    sjsu::LogInfo("left wheel speed: %f", left_wheel_.GetSpeed());
+    sjsu::LogInfo("left wheel position: %f", left_wheel_.GetPosition());
+    sjsu::LogInfo("right wheel speed: %f", right_wheel_.GetSpeed());
+    sjsu::LogInfo("right wheel position: %f", right_wheel_.GetPosition());
+    sjsu::LogInfo("back wheel speed: %f", back_wheel_.GetSpeed());
+    sjsu::LogInfo("back wheel position: %f", back_wheel_.GetPosition());
+    return true;
+  };
```

Comment:
Will probably want to store this to variable for updating server with current rover values